src/parseTools.js

```diff
@@ -176,8 +176,8 @@ function needsQuoting(ident) {
 
 const POINTER_SIZE = MEMORY64 ? 8 : 4;
 const POINTER_BITS = POINTER_SIZE * 8;
-const POINTER_TYPE = 'i' + POINTER_BITS;
-
+const POINTER_TYPE = 'u' + POINTER_BITS;
+const WASM_GLOBAL_POINTER_TYPE = 'i' + POINTER_BITS;
```

Comment:
I don't understand what the variable name here means. Is this a pointer type for a wasm global? Or a global pointer type for wasm?

Why is it different from `POINTER_TYPE`? Maybe I don't understand the motivation here.

Reply:
How about this?
```js
// Similar to POINTER_TYPE, but this is the actual wasm type that is
// used in practice, while POINTER_TYPE is an internal emscripten
// type (that is unsigned, and wasm does not have unsigned types).
var POINTER_WASM_TYPE = 'i32';
```

Reply:
Thanks!